src/github.com/getlantern/lantern-mobile/app/src/main/java/org/getlantern/lantern/activity/LanternMainActivity.java

```diff
@@ -397,13 +404,21 @@ public void onReceive(Context context, Intent intent) {
                 // abruptly closed, we want to clear user preferences
                 Utils.clearPreferences(context);
             } else if (action.equals(ConnectivityManager.CONNECTIVITY_ACTION)) {
-                // whenever a user disconnects from Wifi and Lantern is running
                 NetworkInfo networkInfo =
                     intent.getParcelableExtra(ConnectivityManager.EXTRA_NETWORK_INFO);
-                if (LanternUI.useVpn() && 
-                        networkInfo.getType() == ConnectivityManager.TYPE_WIFI &&
-                        !networkInfo.isConnected()) {
-                    stopLantern();
+
+                if (networkInfo.getType() == ConnectivityManager.TYPE_WIFI) {
+                    if (networkInfo.isConnected()) {
+                        // automatically refresh feed when connectivity is detected
+                        refreshFeed(null);
+                    } else {
+                        // whenever a user disconnects from Wifi and Lantern is running
```

Comment:
That too, but what I was actually saying is that if we disconnect from Wifi but are still connected to a cellular network, we should be allowed to continue browsing as well.